File: engine/src/main/java/com/terralite/engine/entity/EntityManager.java
package com.terralite.engine.entity;

import java.util.Collection;
import java.util.Optional;

public interface EntityManager {
    Entity create();

    Entity add(Entity entity);

    Optional<Entity> get(EntityId id);

    Entity require(EntityId id);

    boolean contains(EntityId id);

    Entity remove(EntityId id);

    Collection<EntityId> ids();

    Collection<Entity> entities();

    int size();
}

File: engine/src/commonMain/kotlin/com/terralite/engine/entity/EntityManager.kt
package com.terralite.engine.entity

interface EntityManager {
    fun create(): Entity
    fun add(entity: Entity): Entity
    fun get(id: EntityId): Entity?
    fun require(id: EntityId): Entity
    fun contains(id: EntityId): Boolean
    fun remove(id: EntityId): Entity
    fun ids(): Collection<EntityId>
    fun entities(): Collection<Entity>
    fun size(): Int
}
